overlord/snapstate/refreshhints.go

```diff
@@ -71,12 +73,25 @@ func (r *refreshHints) refresh() error {
 	perfTimings := timings.New(map[string]string{"ensure": "refresh-hints"})
 	defer perfTimings.Save(r.state)
 
+	var updates []*snap.Info
+	var ignoreValidationByInstanceName map[string]bool
 	timings.Run(perfTimings, "refresh-candidates", "query store for refresh candidates", func(tm timings.Measurer) {
-		_, _, _, err = refreshCandidates(auth.EnsureContextTODO(), r.state, nil, nil, &store.RefreshOptions{RefreshManaged: refreshManaged})
+		updates, _, ignoreValidationByInstanceName, err = refreshCandidates(auth.EnsureContextTODO(), r.state, nil, nil, &store.RefreshOptions{RefreshManaged: refreshManaged})
 	})
 	// TODO: we currently set last-refresh-hints even when there was an
 	// error. In the future we may retry with a backoff.
 	r.state.Set("last-refresh-hints", time.Now())
+
+	if err == nil {
+		deviceCtx, err := DeviceCtxFromState(r.state, nil)
+		if err != nil {
+			return err
+		}
+		hints, err := refreshHintsFromCandidates(r.state, updates, ignoreValidationByInstanceName, deviceCtx)
```

Comment:
And here too ? Error out on err and on no error set refresh-dandidates?

AIUI this and the above change should not change the flow but would be more "regular"?

Reply:
Yeah, not sure why I ended up here... Thanks